cmd/completion.go

```diff
@@ -0,0 +1,63 @@
+package cmd
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+// completionCmd represents the completion command
+var completionCmd = &cobra.Command{
+	Use:   "completion [bash|zsh|fish|powershell]",
+	Short: "Generate a shell completion for Grype (listing local docker images)",
+	Long: `To load completions (docker image list):
+
+Bash:
+
+$ source <(grype completion bash)
+
+# To load completions for each session, execute once:
+Linux:
+  $ grype completion bash > /etc/bash_completion.d/grype
+MacOS:
+  $ grype completion bash > /usr/local/etc/bash_completion.d/grype
+
+Zsh:
+
+# If shell completion is not already enabled in your environment you will need
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ grype completion zsh > "${fpath[1]}/_grype"
+
+# You will need to start a new shell for this setup to take effect.
+
+Fish:
+
+$ grype completion fish | source
+
+# To load completions for each session, execute once:
+$ grype completion fish > ~/.config/fish/completions/grype.fish
+`,
+	DisableFlagsInUseLine: true,
+	ValidArgs:             []string{"bash", "zsh", "fish", "powershell"},
+	Args:                  cobra.ExactValidArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		switch args[0] {
+		case "bash":
+			cmd.Root().GenBashCompletion(os.Stdout)
+		case "zsh":
+			cmd.Root().GenZshCompletion(os.Stdout)
+		case "fish":
+			cmd.Root().GenFishCompletion(os.Stdout, true)
```

Comment:
why does this need `true` here?

Reply:
This was the example Cobra said to use for the generation command so I was following their lead here. 

Upon further inspection of the library code, it looks like this dictates what type of command cobra actually uses to do the shell completion. If it's false, it will use a different command that doesn't include completion result completions? Idk it's a bit confusing:
```go
const (
	// ShellCompRequestCmd is the name of the hidden command that is used to request
	// completion results from the program.  It is used by the shell completion scripts.
	ShellCompRequestCmd = "__complete"
	// ShellCompNoDescRequestCmd is the name of the hidden command that is used to request
	// completion results without their description.  It is used by the shell completion scripts.
	ShellCompNoDescRequestCmd = "__completeNoDesc"
)
```